### ISSUE TITLE
ANN201 false positive when functions are defined within functions

### ISSUE THREAD

**Issue by sumnerevans** [python-discord/flake8-annotations#69]
**Describe the bug**
When an outer function returns `None`, but a function which is defined within it returns something that isn't `None`, the outer function is falsly marked with an `ANN2**` error.

*Edit:* Note this happens even when `suppress-none-returning = True`.

**To Reproduce**
Minimal code example to reproduce the behavior:

```py
def foo():
    def bar() -> int:
        return 1

    print(bar())


foo()
```

When running with `--supress-none-returning`:
```
$ flake8 --suppress-none-returning test.py
test.py:1:11: ANN201 Missing return type annotation for public function
```

**Version Information**
```bash
$ flake8 --version
3.7.9 (flake8-annotations: 2.0.0, flake8-comprehensions: 3.2.2, flake8-print: 3.1.4, flake8_pep3101: 1.2.1, import-order: 0.18.1, mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.1) CPython 3.8.1 on Linux
```

As well as your Python version:
```bash
$ python -V
Python 3.8.1
```

**Comment by sco1** [comment]
This isn’t a false positive, `foo` should be annotated here as it implicitly returns `None`.

**Comment by sumnerevans** [comment]
Sorry, meant to mention that this happens even when `suppress-none-returning = True`

**Comment by sco1** [comment]
Well I know why this is happening but it's looking like the fix is going to be challenging.

While the return node visitor approach is an elegant one for most cases, it breaks for nested functions because it is visiting the return nodes without any knowledge of their containing function. In the code snippet above, for example, the return node ends up being visited twice, once for `bar`, as expected, and the other for `foo`, since the return node is in the body of the `foo` function node. 

Conceptually, this is simply fixed by skipping nested function nodes, but the actual implementation of the fix doesn't seem to be a simple one.

**Comment by sco1** [comment]
Hi @sumnerevans, these issues should be fixed now by `v2.0.1`. Please let us know if we've missed anything!

**Comment by sumnerevans** [comment]
Thanks for getting to this so quickly! Just upgraded, and this issue appears to be fixed.
